test/test_dlpack.py

```diff
@@ -144,6 +144,17 @@ def __dlpack__(self, stream=None):
             x = DLPackTensor(make_tensor((5,), dtype=torch.float32, device=device))
             from_dlpack(x)
 
+    @skipMeta
+    @onlyCUDA
+    @skipCUDAIfRocm
+    def test_dlpack_convert_default_stream(self, device):
+        x = torch.zeros(2**30, device=device)
+        torch.cuda.default_stream().synchronize()
+        self.assertTrue(torch.cuda.default_stream().query())
+        d = x.__dlpack__(1)
+        # check that the default stream has work (a pending cudaStreamWaitEvent)
```

Comment:
why would there be a cudaStreamWaitEvent? Does `__dlpack__` actually do copy?

Reply:
However, looking at this again I believe the `synchronize` is called too late, and I meant to put it before the call to `torch.zeros`.